### PR TITLE
replace pull-kubernetes-e2e-gke-gci with pull-kubernetes-e2e-gke on prow

### PR DESCRIPTION
**Pull request by BenTheElder**
GCI is COS now so -gci doesn't make sense, try to clean this job up and move it to Prow
/area jobs

### CODE REVIEW

**Comment by BenTheElder** [comment]
/cc @spxtr @stevekuznetsov @kargakis 
This is exactly where some form of boilerplate reduction would help :(

> +444 −38

Migrating one presubmit :(

**Comment by BenTheElder** [comment]
@krzyzacy git diff will be borked on this so:
- the prow/config.yaml entries are taken from `pull-kubernetes-e2e-gce` but s/gce/gke, drop (all) from the regex, always_run: true -> false, add `run_if_changed` from the old `pull-kubernetes-e2e-gke-gci`.
- the jobs/config.json entry is based on merging e2e-gce and e2e-gke-gci, I dropped the GCI flag and increased the timeout but otherwise generally preferred flags from the gci job. this may be tricky to review :(

**Comment by BenTheElder** [comment]
> AssertionError: '--stage=gs://kubernetes-release-dev/ci' not found in

@krzyzacy should we still be using that? I have `"--stage=gs://kubernetes-release-pull/ci/pull-kubernetes-e2e-gke"` 😕

**Comment by krzyzacy** [comment]
you can get rid of https://github.com/kubernetes/test-infra/blob/master/jobs/config_test.py#L597-L599

**Comment by BenTheElder** [comment]
> you can get rid of https://github.com/kubernetes/test-infra/blob/master/jobs/config_test.py#L597-L599

done.

File: jobs/config.json [hunk 1]
```diff
@@ -11355,54 +11355,55 @@
...
       "--gke-environment=test",
-      "--gke-shape={\"default\":{\"Nodes\":4,\"MachineType\":\"n1-standard-2\"}}",
-      "--mode=docker",
       "--provider=gke",
-      "--runtime-config=batch/v2alpha1=true,admissionregistration.k8s.io/v1alpha1=true",
       "--stage=gs://kubernetes-release-dev/ci",
-      "--stage-suffix=pull-gke-gci",
-      "--test_args=--ginkgo.flakeAttempts=2 --ginkgo.skip=\\[Slow\\]|\\[Serial\\]|\\[Disruptive\\]|\\[Flaky\\]|\\[Feature:.+\\] --minStartupPods=8",
-      "--timeout=55m"
+      "--stage-suffix=pull-gke-device-plugin-gpu",
```
Comment (krzyzacy) [inline jobs/config.json]: unify the --stage flag here as well

Comment (BenTheElder) [reply jobs/config.json]: done I think?

Comment (krzyzacy) [reply jobs/config.json]: as long as presubmit passes :-)

Comment (BenTheElder) [reply jobs/config.json]: Too many legacy flags 🤷‍♂️

Comment (krzyzacy) [reply jobs/config.json]: we should clean up the code, seems `--stage-suffix` is not used anywhere anymore :-)

Comment (BenTheElder) [reply jobs/config.json]: Will remove it in a follow up, who knows what jobs out there are trying to use this right now.

File: jobs/config.json [hunk 2]
```diff
@@ -11355,54 +11355,54 @@
...
       "--gke-environment=test",
-      "--gke-shape={\"default\":{\"Nodes\":4,\"MachineType\":\"n1-standard-2\"}}",
-      "--mode=docker",
       "--provider=gke",
-      "--runtime-config=batch/v2alpha1=true,admissionregistration.k8s.io/v1alpha1=true",
-      "--stage=gs://kubernetes-release-dev/ci",
-      "--stage-suffix=pull-gke-gci",
-      "--test_args=--ginkgo.flakeAttempts=2 --ginkgo.skip=\\[Slow\\]|\\[Serial\\]|\\[Disruptive\\]|\\[Flaky\\]|\\[Feature:.+\\] --minStartupPods=8",
-      "--timeout=55m"
+      "--stage=gs://kubernetes-release-dev/ci/pull-kubernetes-e2e-gke-device-plugin-gpu",
```
Comment (krzyzacy) [inline jobs/config.json]: `kubernetes-release-pull`, /shrug

Comment (BenTheElder) [reply jobs/config.json]: done

Comment (krzyzacy) [reply jobs/config.json]: feel free to hold cancel whenever, gke master jobs are busted though

Comment (BenTheElder) [reply jobs/config.json]: Might as well get it off of jenkins /shrug

**Pull request merged**